adapters/sovrn/sovrn.go

```diff
@@ -75,71 +100,54 @@ func addHeaderIfNonEmpty(headers http.Header, headerName string, headerValue str
 	}
 }
 
-func (s *SovrnAdapter) MakeBids(internalRequest *openrtb2.BidRequest, externalRequest *adapters.RequestData, response *adapters.ResponseData) (*adapters.BidderResponse, []error) {
-	if response.StatusCode == http.StatusNoContent {
+func (s *SovrnAdapter) MakeBids(request *openrtb2.BidRequest, requestData *adapters.RequestData, responseData *adapters.ResponseData) (*adapters.BidderResponse, []error) {
+	if responseData.StatusCode == http.StatusNoContent {
 		return nil, nil
 	}
 
-	if response.StatusCode == http.StatusBadRequest {
+	if responseData.StatusCode == http.StatusBadRequest {
 		return nil, []error{&errortypes.BadInput{
-			Message: fmt.Sprintf("Unexpected status code: %d. Run with request.debug = 1 for more info", response.StatusCode),
+			Message: fmt.Sprintf("Unexpected status code: %d. Run with request.debug = 1 for more info", responseData.StatusCode),
 		}}
 	}
 
-	if response.StatusCode != http.StatusOK {
+	if responseData.StatusCode != http.StatusOK {
 		return nil, []error{&errortypes.BadServerResponse{
-			Message: fmt.Sprintf("Unexpected status code: %d. Run with request.debug = 1 for more info", response.StatusCode),
+			Message: fmt.Sprintf("Unexpected status code: %d. Run with request.debug = 1 for more info", responseData.StatusCode),
 		}}
 	}
 
-	var bidResp openrtb2.BidResponse
-	if err := json.Unmarshal(response.Body, &bidResp); err != nil {
+	var bidResponse openrtb2.BidResponse
+	if err := json.Unmarshal(responseData.Body, &bidResponse); err != nil {
 		return nil, []error{&errortypes.BadServerResponse{
 			Message: err.Error(),
 		}}
 	}
 
-	bidResponse := adapters.NewBidderResponseWithBidsCapacity(5)
+	response := adapters.NewBidderResponseWithBidsCapacity(5)
 
-	for _, sb := range bidResp.SeatBid {
+	for _, sb := range bidResponse.SeatBid {
 		for i := 0; i < len(sb.Bid); i++ {
 			bid := sb.Bid[i]
 			adm, err := url.QueryUnescape(bid.AdM)
 			if err == nil {
 				bid.AdM = adm
-				bidResponse.Bids = append(bidResponse.Bids, &adapters.TypedBid{
-					Bid:     &bid,
-					BidType: openrtb_ext.BidTypeBanner,
-				})
-			}
-		}
-	}
 
-	return bidResponse, nil
-}
+				bidType := openrtb_ext.BidTypeBanner
 
-func preprocess(imp *openrtb2.Imp) (string, error) {
-	var bidderExt adapters.ExtImpBidder
-	if err := json.Unmarshal(imp.Ext, &bidderExt); err != nil {
-		return "", &errortypes.BadInput{
-			Message: err.Error(),
-		}
-	}
+				if request.Imp[0].Video != nil {
+					bidType = openrtb_ext.BidTypeVideo
+				}
```

Comment:
In case you return bid type in bid.ext we recommend to use this approach:
https://github.com/prebid/prebid-server/blob/c93a0369549f5c087cc41553a31a4bbfe72ddd23/adapters/algorix/algorix.go#L190
It will also help to determine bid type correctly in case if req.imp is multi typed.

Reply:
We do not support multi-typed imps yet.

Reply:
You may not support it, but request may have it. In your code you take imp type from request and match it with response: `request.Imp[impIdx]`

Reply:
You may not support it, but request may have it. In your code you take imp type from request and match it with response: `request.Imp[impIdx]`